ports/stm32/boards/Passport/modules/data_codecs/seedqr_codec.py

```diff
@@ -0,0 +1,62 @@
+# SPDX-FileCopyrightText: © 2020 Foundation Devices, Inc. <hello@foundationdevices.com>
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# seedqr_decoder.py
+#
+# SeedQR codec
+#
+
+from .data_encoder import DataEncoder
+from .data_decoder import DataDecoder
+from .data_sampler import DataSampler
+from .qr_type import QRType
+
+
+class SeedQRDecoder(DataDecoder):
+    def __init__(self):
+        self.data = None
+
+    def add_data(self, data):
+        self.data = data
+
+    def estimated_percent_complete(self):
+        return 1 if self.data is not None else 0
+
+    def is_complete(self):
+        return self.data is not None
+
+    def decode(self, **kwargs):
+        return self.data
+
+    def qr_type(self):
+        return QRType.QR
+
+
+class SeedQREncoder(DataEncoder):
+    def __init__(self, _args):
+        self.data = None
+
+    def encode(self, data, is_binary=False, max_fragment_len=None):
+        import trezorcrypto
+        data_str = ""
+
+        # Get indices in binary string of 11 bits per index
+        for word in data:
```

Comment:
Consider using StringIO here instead of appending (again, it allocated less than `+=`).